Shared/Samples/Trace utility network/TraceUtilityNetworkView.Views.swift

```diff
@@ -0,0 +1,124 @@
+            ForEach([PointType.start, PointType.barrier], id: \.self) { type in
+                Text(type.rawValue.capitalized).tag(type)
+            }
+        }
+        .padding()
+        .pickerStyle(.segmented)
+        Button("Trace") {
+            model.tracingActivity = .tracing
+        }
+        .disabled(model.pendingTraceParameters?.startingLocations.isEmpty ?? true)
+        .padding()
+        Button("Reset", role: .destructive) {
+            model.reset()
+        }
+        .padding()
+    }
+    
+    /// A binding to the type of the starting point being set.
+    ///
+    /// - Note: This should only be used when the user is setting starting points or barriers. If
+    /// this condition isn't present, gets will be inaccurate and sets will be ignored.
+    private var pointType: Binding<PointType> {
+        .init(
+            get: {
+                guard case .settingPoints(let pointType) = model.tracingActivity else {
+                    return .start
+                }
+                return pointType
+            },
+            set: {
+                guard case .settingPoints = model.tracingActivity else { return }
+                model.tracingActivity = .settingPoints(pointType: $0)
+            }
+        )
+    }
+    
+    /// Buttons for each the available terminals on the last added utility element.
+    @ViewBuilder
+    var terminalPickerButtons: some View {
+        ForEach(model.lastAddedElement?.assetType.terminalConfiguration?.terminals ?? []) { terminal in
+            Button(terminal.name) {
+                model.lastAddedElement?.terminal = terminal
+                model.updateUserHint(withMessage: "terminal: \(terminal.name)")
+            }
+        }
+    }
+    
+    /// The menu at the bottom of the screen that guides the user through running a trace.
+    var traceMenu: some View {
+        HStack(spacing: 5) {
+            switch model.tracingActivity {
+            case .none:
+                Button("Start a New Trace") {
+                    withAnimation {
+                        model.traceTypeSelectorIsOpen.toggle()
+                    }
+                }
+                .padding()
+                .confirmationDialog("Trace Type", isPresented: $model.traceTypeSelectorIsOpen) {
+                    traceTypePickerButtons
+                }
+            case .settingPoints:
+                controlsForSettingPoints
+                    .alert(
+                        "Select terminal",
+                        isPresented: $model.terminalSelectorIsOpen,
+                        actions: { terminalPickerButtons }
+                    )
+            case .tracing:
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle())
+                    .padding()
+            case .viewingResults:
+                Button("Reset", role: .destructive) {
+                    model.reset()
```

Comment:
One of the main reasons I put trace type selection first in the workflow was because of the way [UtilityTraceParameter's initializer](https://next.sites.afd.arcgis.com/swift/api-reference/documentation/arcgis/utilitytraceparameters/init(tracetype:startinglocations:)) is.

By choosing the trace type up front we can initializer a trace parameters instance right away with the intended trace type and no additional properties are needed to hold intermediate values.

If we instead defer picking the trace type to later on in the flow, we have to defer the creation of trace parameters and instead hold the chosen starting points and barriers in two additional properties in the model.

In the name of keeping the sample as light as possible while using the API in a way that feels natural (creating parameters right await and then adding barriers and starting points to it) I went this route.

Thoughts on this? I can still change the workflow if you like, I thought it'd just be worth highlighting the downsides first.

Note that a parameters [starting points](https://next.sites.afd.arcgis.com/swift/api-reference/documentation/arcgis/utilitytraceparameters/addstartinglocation(_:)) and [barriers](https://next.sites.afd.arcgis.com/swift/api-reference/documentation/arcgis/utilitytraceparameters/addbarrier(_:)) are mutable where as [trace type](https://next.sites.afd.arcgis.com/swift/api-reference/documentation/arcgis/utilitytraceparameters/tracetype-swift.property) is not.

Reply:
One option would be to keep the workflow as is and add buttons to clear the starting points and barriers.

Reply:
common-samples/pull/3835